src/com/epam/collections/utils/StudentsDoublyLinkedList.java

```diff
@@ -0,0 +1,137 @@
+package com.epam.collections.utils;
+
+import com.epam.collections.interfaces.DoublyLinkedList;
+import com.epam.collections.models.Student;
+
+import java.util.Iterator;
+
+public class StudentsDoublyLinkedList implements DoublyLinkedList, Iterable {
+    private int size = 0;
+    private StudentNode headNode;
+    private StudentNode lastNode;
+
+    public StudentNode getHeadNode() {
+        return headNode;
+    }
+
+    public ListIterator iterator() {
+        return new ListIterator(this);
+    }
+
+    @Override
+    public void push(Student student) {
+        StudentNode node = new StudentNode(student);
+        if (headNode == null) {
+            lastNode = node;
+        } else {
+            node.setNext(headNode);
+            headNode.setPrevious(node);
+        }
+        headNode = node;
+        ++size;
+    }
+
+    @Override
+    public Student pop() {
+        if (headNode == null) {
+            throw new NullPointerException("Students doubly linked list is empty");
```

Comment:
`NullPointerException` does not correctly describe the reason of the exception here.